packages/lodestar/src/chain/validation/attestation.ts

```diff
@@ -172,12 +172,14 @@ export async function validateGossipAttestation(
  * Note: We do not queue future attestations for later processing
  */
 export function verifyPropagationSlotRange(chain: IBeaconChain, attestationSlot: Slot): void {
+  // don't want to have the check for dev network with small SECONDS_PER_SLOT
+  const maximumGossipClockDisparity = chain.config.SECONDS_PER_SLOT <= 3 ? 0 : MAXIMUM_GOSSIP_CLOCK_DISPARITY_SEC;
```

Comment:
Why is this necessary? Does this break the sim tests?

Reply:
yeah otherwise it breaks e2e and sim tests since validators submit attestations too early

Reply:
I understand but it feels too arbitrary to have this here this way. With too early, do you mean the beacon node is busy and it's internal clock is too slow?

Reply:
no, the clock is right but due to the MAXIMUM_GOSSIP_CLOCK_DISPARITY_SEC (0.5s), it can only accept attestations of current slot after `clockSlot + 0.5s` while this PR make validator submit attestations within `clockSlot` and `clockSlot + 0.5s` which is too early

Reply:
@dapplion thanks for pointing that out, it's a bug. I fixed it by implementing `slotWithFutureTolerance` and `slotWithPastTolerance` to make it clear

Reply:
merged via #3409